packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/types/FlashMode.java

```diff
@@ -0,0 +1,16 @@
+package io.flutter.plugins.camera.types;
+
+// Mirrors flash_mode.dart
+public enum FlashMode {
+  off,
+  auto,
+  always;
+
+  public static FlashMode getValueForString(String modeStr) {
+    try {
+      return valueOf(modeStr);
+    } catch (IllegalArgumentException | NullPointerException e) {
+      return null;
+    }
+  }
```

Comment:
This would be good to unit test. We can test all allowed values return the correct enum and make sure that other values return `null`.

Reply:
Agreed! Unit tests added.